cosmos/interchain-security
# test: introduce flag for trace input to e2e framework

## Description

Closes: Part of https://github.com/cosmos/interchain-security/pull/1265



* added a new flag --test-file filename::testrunner
* These changes are made so that the steps and the runner are separated. This is more future proof, and for test-file, specifying the test runner separately is necessary. The test runner for the pre-defined test cases is optional, and if not specified it just uses the first compatible one. I think this will be a useful pattern to have in the future, since we are planning to allow running the same tests in different contexts, e.g. with various configurations

I also refactored the TestRun struct to be called TestConfig - this seems much closer to what it actually represents.
The old name was confusing, because the Steps and the TestRun are different things, adn TestRun really sounds like
it contains the steps, which was not the case.

To review this...
* check that actions*.go, config.go and state.go only contains the refactor of changing TestRun to TestConfig
* check that all references to TestRun are changed to TestConfig
* check the meat of the changes: the behaviour of the new flag in main.go

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...
